data.md

```diff
@@ -52,40 +52,48 @@ These can be used for pattern-matching on the LHS of rules as well (`macro` attr
-    rule maxSInt128      =>  170141183460469231731687303715884105727                                        [macro]  /*   2^127 - 1  */
-    rule minSFixed128x10 => -1701411834604692317316873037158841057280000000000                              [macro]  /* (-2^127    ) * 10^10 */
-    rule maxSFixed128x10 =>  1701411834604692317316873037158841057270000000000                              [macro]  /* ( 2^127 - 1) * 10^10 */
-    rule minSInt256      => -57896044618658097711785492504343953926634992332820282019728792003956564819968  [macro]  /*  -2^255      */
-    rule maxSInt256      =>  57896044618658097711785492504343953926634992332820282019728792003956564819967  [macro]  /*   2^255 - 1  */
-
-    rule minUInt8        =>  0                                                                              [macro]
-    rule maxUInt8        =>  255                                                                            [macro]
-    rule minUInt128      =>  0                                                                              [macro]
-    rule maxUInt128      =>  340282366920938463463374607431768211455                                        [macro]  /*   2^128 - 1  */
-    rule minUFixed128x10 =>  0                                                                              [macro]
-    rule maxUFixed128x10 =>  3402823669209384634633746074317682114550000000000                              [macro]  /* ( 2^128 - 1) * 10^10 */
-    rule minUInt160      =>  0                                                                              [macro]
-    rule maxUInt160      =>  1461501637330902918203684832716283019655932542975                              [macro]  /*   2^160 - 1  */
-    rule minUInt256      =>  0                                                                              [macro]
-    rule maxUInt256      =>  115792089237316195423570985008687907853269984665640564039457584007913129639935 [macro]  /*   2^256 - 1  */
+    rule minSInt ( N )   =>  0 -Int 2 ^Int (N -Int 1)                           [macro]
+    rule maxSInt ( N )   =>  2 ^Int (N -Int 1) -Int 1                           [macro]
+    rule minSInt128 => minSInt ( 128 )                                          [macro]
+    rule maxSInt128 => maxSInt ( 128 )                                          [macro]
+    rule minSFixed128x10 => -1701411834604692317316873037158841057280000000000  [macro]  /* (-2^127    ) * 10^10 */
+    rule maxSFixed128x10 =>  1701411834604692317316873037158841057270000000000  [macro]  /* ( 2^127 - 1) * 10^10 */
+    rule minSInt256 => minSInt ( 256 )                                          [macro]
+    rule maxSInt256 => maxSInt ( 256 )                                          [macro]
+
+    rule minUInt ( N )   =>  0                                                  [macro]
+    rule maxUInt ( N )   =>  2 ^Int (N -Int 1) -Int 1                           [macro]
```

Comment:
2 ^Int N -Int 1

Reply:
@livnev please try this and push, I would like to see (1) if it works, and (2) what effect these changes have on performance.